addons/controls/README.md

```diff
@@ -0,0 +1,419 @@
+
+```jsx
+import { Button } from './Button';
+export default {
+  title: 'Button',
+  component: Button,
+  argTypes: {
+    background: { control: { type: 'color' } },
+  },
+};
+
+export const Basic = (args) => <Button {...args} />;
+Basic.args = { label: 'hello', background: '#ff0' };
+```
+
+This generates the following UI, which is what we wanted in the first place:
+
+<center>
+  <img src="./docs/media/addon-controls-args-background-color.png" width="100%" />
+</center>
+
+### Fully custom args
+
+Up until now, we've only been using auto-generated controls based on the component we're writing stories for. What happens when we want a control for something that's not part of the story?
+
+Consider the following story for our `Button` from above:
+
+```jsx
+import range from 'lodash/range';
+// export default etc.
+
+export const Reflow = ({ count, label, ...args }) => (
+  <>
+    {range(count).map((i) => (
+      <Button label={`${label} ${i}`} {...args} />
+    ))}
+  </>
+);
+Reflow.args = { count: 3, label: 'reflow' };
+```
+
+This generates the following UI:
+
+<center>
+  <img src="./docs/media/addon-controls-args-background-reflow.png" width="100%" />
+</center>
+
+Storybook has inferred the control to be a numeric input based on the initial value of the `count` arg. As we did above, we can also specify a custom control [as we did above](#custom-controls). Only this time since it's story specific we can do it at the story level:
+
+```jsx
+// export const Reflow = ... (as above)
+// Reflow.args = ...
+Reflow.argTypes = {
+  count: { control: { type: 'range', min: 0, max: 20 } },
+};
+```
+
+This generates the following UI with a custom range slider:
+
+<center>
+  <img src="./docs/media/addon-controls-args-background-reflow-slider.png" width="100%" />
+</center>
+
+### Template stories
+
+Suppose you've created the `Basic` story from above, but now we want to create a second story with a different state, such as how the button renders with the label is really long.
+
+The simplest thing would be to create a second story:
+
+```jsx
+export const VeryLongLabel = (args) => <Button {...args} />;
+VeryLongLabel.args = { label: 'this is a very long string', background: '#ff0' };
+```
+
+This works, but it repeats code. What we really want is to reuse the `Basic` story, but simply create a different initial state. In Storybook we do this idiomatically for Args stories:
+
+```jsx
+export const VeryLongLabel = Basic.bind();
+VeryLongLabel.args = { label: 'this is a very long string', background: '#ff0' };
+```
+
+We can even reuse initial args from other stories:
+
+```jsx
+export const VeryLongLabel = Basic.bind();
+VeryLongLabel.args = { ...Basic.args, label: 'this is a very long string' };
+```
+
+<center>
+  <img src="./docs/media/addon-controls-args-template.png" width="100%" />
+</center>
+
+## Migrating from knobs
```

Comment:
I suggest we move this section to the FAQ. Something like "How to migrate from knobs". This isn't necessary for folks getting started with Controls and will quickly lose relevance as controls is widely adopted (which means we'll have to move it anyways).